pkgs/applications/window-managers/sway/wrapper.nix

```diff
@@ -16,7 +16,7 @@ assert extraSessionCommands != "" -> withBaseWrapper;
 with lib;
 
 let
-  sway = sway-unwrapped.override { inherit isNixOS enableXWayland; };
+  sway = sway-unwrapped.overrideAttrs (oa: { inherit isNixOS enableXWayland; });
```

Comment:
Hm, I would've kept the old code here. IMO `override` is preferable but I've never looked into the technical details and the documentation doesn't make any such statement either so `overrideAttrs` is also fine with me. Enabling both `override` and `overrideAttrs` makes things a bit "confusing" though (but dropping the package arguments probably causes unnecessary trouble for users who `override` Sway.

Reply:
I dont know if one is preferred. I've kept the variables in `override` not to break existing setups, also it makes the configuration knobs easier to spot. I find the use of overrideAttrs much easier to work with though because you can explore them via the repl, dump them etc. In the scenario of home-manager, I want to replace the default "sway" package with "swayfx". If I keep the `override`, then home-manager will try to override the settings of swayfx which doesn't accept the same inputs as sway-unwrapped and thus it is not possible. `overrideAttrs` ends up modifying the final derivation and works in both cases so I would say it's better from my point of view ^^ . Let me know if you are still ok to merge, dont want to go against the maintainers

Reply:
Ok, it would be great if we could get input from someone who's really familiar with the technical differences but it should be fine/acceptable either way so I'm fine with merging it as it is.